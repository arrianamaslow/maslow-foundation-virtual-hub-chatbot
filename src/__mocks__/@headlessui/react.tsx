/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
