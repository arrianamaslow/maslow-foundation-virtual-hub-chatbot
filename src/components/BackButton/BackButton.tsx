import { useRouter } from 'next/navigation'
import React from 'react'

export type BackButtonProps = {
  href: string
}

const BackButton: React.FC<BackButtonProps> = ({ href }) => {
  const router = useRouter()
  return (
    <button
      data-testid='back-button'
      onClick={() => {
        router.push(href)
      }}
      className='h-6 w-16'
    >
      <div className='flex flex-row'>
        <img
          className='border-[1.5px] border-black rounded-lg py-[6px] w-6 h-6'
          src='/img/back-arrow.svg'
        />
        <span className='flex justify-center text-base pl-[7px] font-baskervald'>Back</span>
      </div>
    </button>
  )
}
export default BackButton
