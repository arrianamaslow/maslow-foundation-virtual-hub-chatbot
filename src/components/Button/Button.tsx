import style from './style.module.css'
export type ButtonProps = {
  text: string
  isDisabled?: boolean
  onClick: () => void
  buttonType: 'primary' | 'secondary' | 'tertiary'
}
const Button: React.FC<ButtonProps> = ({ text, isDisabled, onClick, buttonType }) => {
  return (
    <button
      className={`inline-flex w-full justify-center px-[24px] py-[10px] flex-shrink-0 rounded-lg border-solid border-[1px] text-[16px] ${style[buttonType]}`}
      onClick={onClick}
      disabled={isDisabled}
    >
      {text}
    </button>
  )
}
export default Button
