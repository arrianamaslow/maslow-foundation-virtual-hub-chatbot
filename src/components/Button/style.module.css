.primary {
  @apply border-black bg-black text-white hover:border-gray-500 hover:bg-gray-500 active:bg-stone-500 active:border-black;
}

.primary:disabled {
  @apply border-stone-300 bg-stone-300 text-white;
}

.secondary {
  @apply border-black bg-white hover:border-stone-300 active:border-black;
}

.secondary:disabled {
  @apply border-stone-300 bg-white text-stone-300;
}

.tertiary {
  @apply border-white bg-white hover:underline hover:text-stone-300 active:text-black active:underline;
}

.tertiary:disabled {
  @apply border-white bg-white text-stone-300;
}
