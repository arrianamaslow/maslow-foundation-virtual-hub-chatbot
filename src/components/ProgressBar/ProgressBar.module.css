.doneSegment {
  @apply h-[8px] grow bg-black;
}

.notDoneSegment {
  @apply h-[8px] grow bg-[#D9D9D9];
}

.segmentContainer {
  @apply flex gap-x-[14px];
}
