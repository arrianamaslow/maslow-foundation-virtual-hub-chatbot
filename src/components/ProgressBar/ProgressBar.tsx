import React from 'react'
import style from './ProgressBar.module.css'

export type ProgressBarProps = {
  totalSegments: number
  currentSegment: number
}

const ProgressBar: React.FC<ProgressBarProps> = ({ totalSegments, currentSegment }) => {
  return (
    <div className={style.segmentContainer}>
      {Array(totalSegments)
        .fill(1)
        .map((e, i) => (
          <div key={i} className={i < currentSegment ? style.doneSegment : style.notDoneSegment} />
        ))}
    </div>
  )
}

export default ProgressBar
