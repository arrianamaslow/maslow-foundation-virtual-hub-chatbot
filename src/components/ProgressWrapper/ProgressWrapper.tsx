import React, { ReactNode } from 'react'
import BackButton from '@/components/BackButton/BackButton'
import ProgressBar from '@/components/ProgressBar/ProgressBar'
import Button from '@/components/Button/Button'
export type ProgressWrapperProps = {
  stepHeaderText: string
  backButtonUrl: string
  currentProgressSegment: number
  totalProgressSegments: number
  isContinueDisabled: boolean
  onContinueClicked: () => void
  children: ReactNode
}
const ProgressWrapper: React.FC<ProgressWrapperProps> = ({
  stepHeaderText,
  onContinueClicked,
  backButtonUrl,
  currentProgressSegment,
  totalProgressSegments,
  isContinueDisabled: isDisabled,
  children
}) => {
  return (
    <div className='h-full flex flex-col w-full md:pb-[64px] md:px-[83px] px-[16px] py-[32px] bg-white md:rounded-xl md:shadow-xl '>
      <div className='h-[40px] hidden md:block'>
        {currentProgressSegment !== 1 && <BackButton href={backButtonUrl} />}
      </div>
      <div className='flex flex-col gap-y-[24px] md:gap-y-[30px] grow'>
        <h3 className='font-baskervald text-[20px] md:text-[38px]' data-testid='step-header-text'>
          {stepHeaderText}
        </h3>
        <ProgressBar
          totalSegments={totalProgressSegments}
          currentSegment={currentProgressSegment}
        />
        <div className='flex grow'>{children}</div>
        <div className='grow md:hidden'></div>
        <div className='flex self-end md:self-auto'>
          <Button
            text={'Continue'}
            buttonType={'primary'}
            isDisabled={isDisabled}
            onClick={onContinueClicked}
          />
        </div>
      </div>
    </div>
  )
}
export default ProgressWrapper
