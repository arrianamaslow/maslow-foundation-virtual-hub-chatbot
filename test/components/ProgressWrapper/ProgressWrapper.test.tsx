/**
 * @jest-environment jsdom
 */

import ProgressWrapper, { ProgressWrapperProps } from '@/components/ProgressWrapper/ProgressWrapper'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { BackButtonProps } from '@/components/BackButton/BackButton'
import { ProgressBarProps } from '@/components/ProgressBar/ProgressBar'
import { ButtonProps } from '@/components/Button/Button'

const defaultProps: ProgressWrapperProps = {
  children: <p data-testid='test-child-component'>CHILDREN</p>,
  stepHeaderText: 'test title',
  backButtonUrl: '/back-url',
  currentProgressSegment: 4,
  totalProgressSegments: 4,
  isContinueDisabled: false,
  onContinueClicked: jest.fn()
}

jest.mock('@/components/BackButton/BackButton', () => {
  const MockBackButton = ({ href }: BackButtonProps) => (
    <div>
      <p data-testid='back-button-href'>{href}</p>
    </div>
  )
  return MockBackButton
})

jest.mock('@/components/ProgressBar/ProgressBar', () => {
  const MockProgressBar = ({ totalSegments, currentSegment }: ProgressBarProps) => (
    <div>
      <p data-testid='progress-bar-currentSegment'>{currentSegment}</p>
      <p data-testid='progress-bar-totalSegments'>{totalSegments}</p>
    </div>
  )
  return MockProgressBar
})

jest.mock('@/components/Button/Button', () => {
  const MockButton = ({ isDisabled, onClick, text }: ButtonProps) => (
    <div>
      <button data-testid='button-clickable' onClick={onClick}>
        {text}
      </button>
      <p data-testid='button-isDisabled'>{`${isDisabled}`}</p>
    </div>
  )
  return MockButton
})

describe('<ProgressWrapper />', () => {
  it('renders as expected', () => {
    const tree = render(<ProgressWrapper {...defaultProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders children as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const child = screen.getByTestId('test-child-component')
    expect(child).toHaveTextContent(`CHILDREN`)
  })

  it('renders stepHeaderText as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const textContainer = screen.getByTestId('step-header-text')
    expect(textContainer).toHaveTextContent(defaultProps.stepHeaderText)
  })

  it('passes href to <BackButton /> as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const textContainer = screen.getByTestId('back-button-href')
    expect(textContainer).toHaveTextContent(defaultProps.backButtonUrl)
  })

  it('passes currentSegment to <ProgressBar /> as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const textContainer = screen.getByTestId('progress-bar-currentSegment')
    expect(textContainer).toHaveTextContent(`${defaultProps.currentProgressSegment}`)
  })

  it('passes totalSegments to <ProgressBar /> as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const textContainer = screen.getByTestId('progress-bar-totalSegments')
    expect(textContainer).toHaveTextContent(`${defaultProps.totalProgressSegments}`)
  })

  it('passes isDisabled to <Button /> as expected', () => {
    render(<ProgressWrapper {...defaultProps} />)
    const textContainer = screen.getByTestId('button-isDisabled')
    expect(textContainer).toHaveTextContent(`${defaultProps.isContinueDisabled}`)
  })

  it('fires onContinueClicked on button click', () => {
    const mockEvent = jest.fn()
    render(<ProgressWrapper {...defaultProps} onContinueClicked={mockEvent} />)

    const clickableButton = screen.getByTestId('button-clickable')
    fireEvent.click(clickableButton)
    expect(mockEvent).toHaveBeenCalled()
  })
})
