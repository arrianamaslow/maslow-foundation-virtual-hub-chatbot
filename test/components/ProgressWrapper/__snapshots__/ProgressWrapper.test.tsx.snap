// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressWrapper /> renders as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-full flex flex-col w-full md:pb-[64px] md:px-[83px] px-[16px] py-[32px] bg-white md:rounded-xl md:shadow-xl "
      >
        <div
          class="h-[40px] hidden md:block"
        >
          <div>
            <p
              data-testid="back-button-href"
            >
              /back-url
            </p>
          </div>
        </div>
        <div
          class="flex flex-col gap-y-[24px] md:gap-y-[30px] grow"
        >
          <h3
            class="font-baskervald text-[20px] md:text-[38px]"
            data-testid="step-header-text"
          >
            test title
          </h3>
          <div>
            <p
              data-testid="progress-bar-currentSegment"
            >
              4
            </p>
            <p
              data-testid="progress-bar-totalSegments"
            >
              4
            </p>
          </div>
          <div
            class="flex grow"
          >
            <p
              data-testid="test-child-component"
            >
              CHILDREN
            </p>
          </div>
          <div
            class="grow md:hidden"
          />
          <div
            class="flex self-end md:self-auto"
          >
            <div>
              <button
                data-testid="button-clickable"
              >
                Continue
              </button>
              <p
                data-testid="button-isDisabled"
              >
                false
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-full flex flex-col w-full md:pb-[64px] md:px-[83px] px-[16px] py-[32px] bg-white md:rounded-xl md:shadow-xl "
    >
      <div
        class="h-[40px] hidden md:block"
      >
        <div>
          <p
            data-testid="back-button-href"
          >
            /back-url
          </p>
        </div>
      </div>
      <div
        class="flex flex-col gap-y-[24px] md:gap-y-[30px] grow"
      >
        <h3
          class="font-baskervald text-[20px] md:text-[38px]"
          data-testid="step-header-text"
        >
          test title
        </h3>
        <div>
          <p
            data-testid="progress-bar-currentSegment"
          >
            4
          </p>
          <p
            data-testid="progress-bar-totalSegments"
          >
            4
          </p>
        </div>
        <div
          class="flex grow"
        >
          <p
            data-testid="test-child-component"
          >
            CHILDREN
          </p>
        </div>
        <div
          class="grow md:hidden"
        />
        <div
          class="flex self-end md:self-auto"
        >
          <div>
            <button
              data-testid="button-clickable"
            >
              Continue
            </button>
            <p
              data-testid="button-isDisabled"
            >
              false
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
